index.js

```diff
@@ -1,21 +1,21 @@
-modules.exports = function redirectToHTTPS (ignoreHosts, ignoreRoutes) {
-  if (!ignoreHosts) {
-    ignoreHosts = [];
-  }
+function searchIgnore (value, ignores) {
+  return ignores.some((search) => {
+    return search.test(value)
+  })
+}
 
-  if (!ignoreRoutes) {
-    ignoreRoutes = [];
-  }
+function redirectToHTTPS (ignoreHosts = [], ignoreRoutes = []) {
+  return function middlewareRedirectToHTTPS (req, res, next) {
+    const isNotSecure = (!req.get('x-forwarded-port') && req.protocol !== 'https') ||
```

Comment:
This conditional line doesn't look right? 🤔 

If there's no `x-forwarded-port` data, but you receive a `https` request, It's supposed to be secured, thus, `isNotSecure` should be `false`, but you'd get an evaluation as follow:
```
req.get('x-forwarded-port') // undefined
req.protocol !== 'https' // false

// thus:

(!req.get('x-forwarded-port') && req.protocol !== 'https')
(!undefined && false)
(true && false) // false

// and then:
parseInt(req.get('x-forwarded-port'), 10)
parseInt(undefined, 10) // NaN

// NaN compared to anything returns false, thus
NaN !== 443 // true

// and all together, isNotSecure is computed as:
false || true // true, but it should be false
```